src/prefect/concurrency/_asyncio.py

```diff
@@ -58,6 +68,90 @@ async def aacquire_concurrency_slots(
+) -> ConcurrencyLimitWithLeaseResponse:
+    try:
+        # Use a run logger if available
+        logger = get_run_logger()
+    except Exception:
+        logger = get_logger("concurrency")
+
+    try:
+        with timeout_async(seconds=timeout_seconds):
+            async with get_client() as client:
+                while True:
+                    try:
+                        response = await client.increment_concurrency_slots_with_lease(
+                            names=names,
+                            slots=slots,
+                            mode=mode,
+                            lease_duration=lease_duration,
+                        )
+                        retval = ConcurrencyLimitWithLeaseResponse.model_validate(
+                            response.json()
+                        )
+                        if not retval.limits:
+                            if strict:
+                                raise ConcurrencySlotAcquisitionError(
+                                    f"Concurrency limits {names!r} must be created before acquiring slots"
+                                )
+                            else:
+                                logger.warning(
+                                    f"Concurrency limits {names!r} do not exist - skipping acquisition."
+                                )
+
+                        return retval
+                    except httpx.HTTPStatusError as exc:
+                        if not exc.response.status_code == 423:  # HTTP_423_LOCKED
+                            raise
+
+                        if max_retries is not None and max_retries <= 0:
+                            raise exc
+                        retry_after = float(exc.response.headers["Retry-After"])
+                        logger.debug(
+                            f"Unable to acquire concurrency slot. Retrying in {retry_after} second(s)."
+                        )
+                        await asyncio.sleep(retry_after)
+                        if max_retries is not None:
+                            max_retries -= 1
+    except TimeoutError as timeout:
+        raise AcquireConcurrencySlotTimeoutError(
+            f"Attempt to acquire concurrency slots timed out after {timeout_seconds} second(s)"
+        ) from timeout
+    except Exception as exc:
+        raise ConcurrencySlotAcquisitionError(
+            f"Unable to acquire concurrency slots on {names!r}"
+        ) from exc
+
+
+async def amaintain_concurrency_lease(
+    lease_id: UUID,
+    lease_duration: float,
+) -> None:
+    """
+    Maintain a concurrency lease by renewing it after the given interval.
+
+    Args:
+        lease_id: The ID of the lease to maintain.
+        lease_duration: The duration of the lease in seconds.
+    """
+    async with get_client() as client:
+        while True:
+            await asyncio.sleep(  # Renew the lease 3/4 of the way through the lease duration
+                lease_duration * 0.75
```

Comment:
why 3/4?

Reply:
Just to give enough time for renewal and make sure that the lease doesn't accidentally expire if retries are needed at the network level.